### PR TITLE
input to tokens

### LINKED ISSUE #8: Tokenizer

### PR DESCRIPTION
**Pull request by pbateman828**
Creates an iterator based on whitespace of the str given.
Fixes #8

### CODE REVIEW

File: src/interpreter/README [hunk 1]
```diff
@@ -0,0 +1 @@
+Beginning of tokenizer.
```
Comment (dmadisetti) [inline src/interpreter/README]: no need for readme

File: src/interpreter/token.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// split to iterator.
+use std::str;
+
+type MyIter<'a> = str::SplitWhitespace<'a>; //type aliasing
```
Comment (dmadisetti) [inline src/interpreter/token.rs]: MyIter -> TokenIterator
comments should also have whitespace. e.g. `// type aliasing` vs. `//type aliasing`, and be complete sentences used for clarification.

File: src/interpreter/token.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// split to iterator.
```
Comment (dmadisetti) [inline src/interpreter/token.rs]: Descriptions should be commented as `//!` and be complete sentences.

File: src/interpreter/token.rs [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+use std::str;
+
+type MyIter<'a> = str::SplitWhitespace<'a>; //type aliasing
+
+pub fn token(text: &str) -> MyIter {
```
Comment (dmadisetti) [inline src/interpreter/token.rs]: `token` -> `tokenize`

File: src/interpreter/token.rs [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    text.split_whitespace()
```
Comment (dmadisetti) [inline src/interpreter/token.rs]: add return for consistency

File: src/lib.rs [hunk 6]
```diff
@@ -6,3 +6,4 @@ pub mod protos;
 pub mod screen;
 pub mod io;
 pub mod load;
+pub mod interpreter;
```
Comment (dmadisetti) [inline src/lib.rs]: alphabetize while you're here :)

File: src/main.rs [hunk 7]
```diff
@@ -13,6 +13,8 @@ use maeve::load::new;
 use maeve::protos::master::Game;
 use maeve::screen::Interfaceable;
 use maeve::screen::Screen;
+use maeve::interpreter::token::token;
```
Comment (dmadisetti) [inline src/main.rs]: here too :D

File: src/main.rs [hunk 8]
```diff
@@ -13,6 +13,8 @@ use maeve::load::new;
...
+
```
Comment (dmadisetti) [inline src/main.rs]: nit: remove superfluous whitespace

File: src/main.rs [hunk 9]
```diff
@@ -43,11 +45,17 @@ fn main() {
         (_, _) => prompt(&mut src),
     };
 
+
```
Comment (dmadisetti) [inline src/main.rs]: nit: whitespace

File: src/main.rs [hunk 10]
```diff
@@ -43,11 +45,17 @@ fn main() {
...
     match result {
         Ok(_game) => {
             src.print("And the games begin!"); // Do something with the games here.
-                                               //Call the interpreter
-                                               //derpreter(game);
+            println!("Where would you like to go?"); //Call the interpreter. derpreter(game);
```
Comment (dmadisetti) [inline src/main.rs]: use src.print

File: src/main.rs [hunk 11]
```diff
@@ -43,11 +45,17 @@ fn main() {
...
+
+            let test_token = src.prompt();   //testing to see if token does what it's supposed to
+            let test_tokens = token(&test_token);
```
Comment (dmadisetti) [inline src/main.rs]: let tokens = tokenize(&scr.prompt())

File: src/main.rs [hunk 12]
```diff
@@ -43,11 +45,17 @@ fn main() {
...
+            for val in test_tokens {
+                println!("Iterator {}", val);
```
Comment (dmadisetti) [inline src/main.rs]: src.print

File: src/main.rs (hunk 10 above)
Comment (dmadisetti) [inline src/main.rs]: Let's move this block out of main into `evaluate.rs` or something. Ideally we want to put as little logic as possible in this file, and leave most of it within the lib.

**Review by dmadisetti — CHANGES_REQUESTED** [review]
all in the spirit of growth

**Comment by dmadisetti** [comment]
Also for good luck, run `cargo fmt`

https://github.com/rust-lang-nursery/rustfmt

**Comment by kevinlmadison** [comment]
Top of the line

File: src/interpreter/token.rs (hunk 5 above)
Comment (kevinlmadison) [reply src/interpreter/token.rs]: I would say we should remove returns in other places for consistency so as to conform to standard Rust style.

**Pull request merged**